File: code/30132.pas
Program Bai132;
var n, c, d :integer;
begin
  readln(n);
  d := n mod 10;
  while n > 0 do 
    begin
      c := n mod 10;
      if c <> d then
       begin
        writeln('NO');
        exit;
       end;
       n := n div 10;
	 end;
  writeln ('YES');
end.

File: code/30132.cpp
#include <iostream>
using namespace std;

int main() 
{
    int n,d,c;
    cin>>n;
    d=n%10;
    while (n>0){
        c=n%10;
        if (c!=d){
            cout<<"NO";
            return 0;
        }
        n=n/10;
    }
    cout<<"YES";
    return 0;
}